File: java/Trees/Binary_Tree_Traversals/Preorder_Traversal.java
package java.Trees.Binary_Tree_Traversals;

import java.util.*;

public class Preorder_Traversal {
  List<Integer> result = new ArrayList<>();

  public List<Integer> preorderTraversal(TreeNode root) {
    preorder(root);
    return result;
  }

  void preorder(TreeNode node) {
    if (node == null)
      return;
    result.add(node.val);
    preorder(node.left);
    preorder(node.right);
  }
}

File: python/Tree/Binary Tree Traversals/Preorder_Traversal.py
class Solution:
  def preordertraversal(self, root):
    result = []

    def dfs(node):
      if not node:
        return 
      
      result.append(node.val)
      dfs(node.left)
      dfs(node.right)
      
    dfs(root)
    return result